### PR TITLE
feat: Add two new flags to the bootstrap command

### PR DESCRIPTION
**Pull request by Naumel**
----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts [hunk 1]
```diff
@@ -76,14 +76,18 @@ export class Bootstrapper {
...
-
-    const current = await BootstrapStack.lookup(sdkProvider, environment, options.toolkitStackName);
-
     if (params.createCustomerMasterKey !== undefined && params.kmsKeyId) {
       throw new Error('You cannot pass \'--bootstrap-kms-key-id\' and \'--bootstrap-customer-key\' together. Specify one or the other');
     }
 
+    if (!!params.defaultPermissionsBoundary === !!params.customPermissionBoundary) {
+      throw new Error('You cannot pass \'--create-default-permissions-boundary\' and \'--permissions-boundary\' together. Specify one or the other');
+    }
```
Comment (Naumel) [inline packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts:85]: If we're going to end up throwing an error, might as well do so fast.

**Review by corymhall — COMMENTED** [review]
Just a couple of minor comments.

**Comment by Naumel** [comment]
@mergifyio update

**Comment by Naumel** [comment]
@mergifyio update

**Comment by Naumel** [comment]
@Mergifyio update

**Comment by Naumel** [comment]
@Mergifyio update

**Comment by Naumel** [comment]
Moving to 22792 since the commits became a tangled mess,.

**Pull request closed without merging**